modules/programs/i3lock.nix

```diff
@@ -0,0 +1,291 @@
+    (optionalString (cfg.colors != null && cfg.colors.layoutcolor != null)
+      "--layoutcolor=${cfg.colors.layoutcolor}")
+    (optionalString (cfg.colors != null && cfg.colors.keyhlcolor != null)
+      "--keyhlcolor=${cfg.colors.keyhlcolor}")
+    (optionalString (cfg.colors != null && cfg.colors.bshlcolor != null)
+      "--bshlcolor=${cfg.colors.bshlcolor}")
+  ]);
+
+  lockScript = pkgs.writeScript "i3lock.sh" ''
+    #!${pkgs.stdenv.shell}
+    ${cfg.extraCommand}
+    ${lockCmd}
+  '';
+in {
+  options.programs.i3lock = {
+    enable = mkEnableOption "i3lock: a simple screen locker like slock.";
+
+    package = mkOption {
+      description = "Package to use for i3lock";
+      type = types.package;
+      default = if enablei3LockColor then pkgs.i3lock-color else pkgs.i3lock;
+    };
+
+    color = mkOption {
+      description = "Background color to use for i3lock.";
+      type = types.nullOr types.str;
+      default = null;
+    };
+
+    beep = mkOption {
+      description = "Whether to enable i3lock beeping";
+      type = types.bool;
+      default = false;
+    };
+
+    pointer = mkOption {
+      description = "Whether to show mouse pointer, or display a hardcoded Windows-Pointer";
+      default = null;
+      type = types.nullOr (types.enum ["default" "win"]);
+    };
+
+    image = mkOption {
+      description = "Background image file to use for i3lock.";
+      type = types.nullOr types.path;
+      default = null;
+    };
+
+    blur = mkOption {
+      description = "Blur the current screen and use that as a background.";
+      type = types.nullOr types.int;
+      default = null;
+    };
+
+    clock = {
+      enable = mkEnableOption "show clock on i3lock.";
+
+      timeStr = mkOption {
+        description = "i3lock clock time format string.";
+        type = types.str;
+        default = "%H:%M:%S";
+      };
+
+      dateStr = mkOption {
+        description = "i3lock clock date format string.";
+        type = types.str;
+        default = "%A, %m %Y";
+      };
+
+      timeFont = mkOption {
+        description = "i3lock font to use for diplaying time.";
+        type = types.nullOr types.str;
+        default = null;
+      };
+
+      dateFont = mkOption {
+        description = "i3lock font to use for diplaying date.";
+        type = types.nullOr types.str;
+        default = null;
+      };
+    };
+
+    verifyText = mkOption {
+      description = "i3lock text dispayed if user is verified.";
+      type = types.nullOr types.str;
+      default = null;
+      example = "Drinking verification can...";
+    };
+
+    wrongText = mkOption {
+      description = "i3lock text displayed if verification fails.";
+      type = types.nullOr types.str;
+      default = null;
+      example = "Nope!";
+    };
+
+    textSize = mkOption {
+      description = "Size of the text";
+      type = types.nullOr types.int;
+      default = null;
+    };
+
+    modSize = mkOption {
```

Comment:
Doesn't seem like this option is used anywhere and I can't find it in manual.

Reply:
Removed